src/cls/rbd/cls_rbd.cc

```diff
@@ -5073,7 +5073,8 @@ int image_status_set(cls_method_context_t hctx, const string &global_image_id,
   if (r < 0) {
     return 0;
   }
-  if (mirror_image.state != cls::rbd::MIRROR_IMAGE_STATE_ENABLED) {
+  if (mirror_image.state != cls::rbd::MIRROR_IMAGE_STATE_ENABLED &&
+      mirror_image.state != cls::rbd::MIRROR_IMAGE_STATE_CREATING) {
```

Comment:
I'm looking at the commit that made this change https://github.com/ceph/ceph/pull/41696/commits/416e25794af0252ec45b35c897c8cf2e91aca383
The commit message says "This prevent image_status_set to succeed when there is no mirror image
yet". 

I'm wondering whether the prior `if` statement is sufficient. If the `image_get` of the mirror image doesn't succeed i.e. mirror image state is disabled and mirroring image is not available, we don't proceed to set the status.  Do we need this `if` statement that checks for statuses at all?

Reply:
>  Do we need this if statement that checks for statuses at all?

Did you mean states instead of statuses?  Superficially, the state check still makes sense to me because it excludes `MIRROR_IMAGE_STATE_DISABLING` which is when image status can be getting removed -- see how `image_status_remove()` is called from `image_remove()`.

Reply:
Oops, I meant states. 

Got it. We want to return early even during MIRROR_IMAGE_STATE_DISABLING state. Thanks!